VertexCompositeProducer/src/LamC3PFitter.cc

```diff
@@ -350,24 +395,31 @@ void LamC3PFitter::fitLamCCandidates(
 //      if( (theTrackRefs[trdx1]->pt() + theTrackRefs[trdx2]->pt()) < tkPtSumCut) continue;
 //      if( abs(theTrackRefs[trdx1]->eta() - theTrackRefs[trdx2]->eta()) > tkEtaDiffCut) continue;
 
+      std::map<std::string, float> theMTDtrackInfo1 = theMTDtrackInfo_sgn1[trdx1]; // mtd
+      std::map<std::string, float> theMTDtrackInfo2 = theMTDtrackInfo_sgn1[trdx2]; // mtd
+      bool isPion1 = theMTDtrackInfo1["isPion"];
+      bool isProton1 = theMTDtrackInfo1["isProton"];
+      bool isPion2 = theMTDtrackInfo2["isPion"];
+      bool isProton2 = theMTDtrackInfo2["isProton"];
+
+      if(!((isPion1&&isProton2) || (isPion2&&isProton1))) continue;
+
       //This vector holds the pair of oppositely-charged tracks to be vertexed
       std::vector<TransientTrack> transTracks;
 
       TrackRef trackRef1 = theTrackRefs_sgn1[trdx1];
       TrackRef trackRef2 = theTrackRefs_sgn1[trdx2];
       TransientTrack* transTkPtr1 = &theTransTracks_sgn1[trdx1];
       TransientTrack* transTkPtr2 = &theTransTracks_sgn1[trdx2];
-      std::map<std::string, float> theMTDtrackInfo1 = theMTDtrackInfo_sgn1[trdx1]; // mtd
-      std::map<std::string, float> theMTDtrackInfo2 = theMTDtrackInfo_sgn1[trdx2]; // mtd
 
       TLorentzVector track1Mom, track2Mom;
       track1Mom.SetPtEtaPhiM(trackRef1->pt(), trackRef1->eta(), trackRef1->phi(), protonMassLamC3PSquared);
-      track2Mom.SetPtEtaPhiM(trackRef2->pt(), trackRef2->eta(), trackRef2->phi(), kaonMassLamC3PSquared);
```

Comment:
The mass here should be:
kaonMassLamC3P  instead of kaonMassLamC3PSquared
protonMassLamC3P instead of protonMassLamC3PSquared